### ISSUE TITLE
Decorator method call from assosiation 

### ISSUE THREAD

**Issue by paxer** [amatsuda/active_decorator#26]
It looks like when you have a situation: 

``` ruby
class Order < ActiveRecord::Base
  has_one :coupon
end

class Coupon < ActiveRecord::Base
  belongs_to :order
end

module CouponDecorator
  def type_description
    #some logic here
  end
end
```

Now when in the view I call 

``` ruby
<%= order.coupon.type_description %>
```

I've got NoMethod exception, but if I call in the another view

``` ruby
<%= coupon.type_description %>
```

it works fine.

The only one workaround I found for now is to create a method in Coupon model  with the same name as decorator 

``` ruby
 def type_description
    ActiveDecorator::Decorator.instance.decorate(self).type_description
 end
```

and now it will work with the calls such as

 <%= order.coupon.type_description %>

but it does not looks good to me, is there any better solution?

**Comment by AlexanderFedorenko** [comment]
Same problem

**Comment by sockmonk** [comment]
The docs suggest moving the coupons to a partial, passing order.coupons as local. Looks reasonable, as it also avoids a Law Of Demeter issue.

**Comment by thutterer** [comment]
> The docs suggest moving the coupons to a partial, passing order.coupons as local. Looks reasonable, as it also avoids a Law Of Demeter issue.

https://github.com/amatsuda/active_decorator#decorating-associated-objects

**Comment by amatsuda** [comment]
This feature has been implemented in #8 and already available in recent releases.
